utils/zerovec/src/varzerovec/owned.rs

```diff
@@ -90,6 +97,51 @@ impl<T: AsVarULE> VarZeroVecOwned<T> {
         self.get_components().get(idx)
     }
 
+    /// Get the position of a specific element in the data segment.
```

Comment:
I think for most of these functions `entire_slice` has to be entirely valid, except for `index_data()` where only the indices array needs to be in the right position (but may have incorrect indices) and `set_len()` where nothing needs to be valid

Reply:
Done, not sure how strictly worded the safety section needs to be here

Reply:
I think include something about how `entire_slice()` needs to be well-formed. Because most of the time we're operating on transitional states of `entire_slice()`, these methods can't be called then.

For example, in `shift_indices()` it's actually valid to call that on a buffer with a malformed index segment, but it must have the correct length set (and that's when we call it).

Reply:
I'm not sure how much better I can do here without precisely explaining all invariants of the slice where the helper functions are used.